addons/base_vat/models/res_partner.py

```diff
@@ -226,6 +226,9 @@ def check_vat_ie(self, vat):
                                    br"[ \-_]?" \
                                    br"(?P<code>[A-Za-z0-9&\xd1\xf1]{3})$")
 
+    # Netherlands VAT verification
+    __check_vat_nl_re = re.compile("(?:NL)?[0-9]{9}B[0-9]{2}")
```

Comment:
> Op termijn kan de structuur met uitzondering van de landaanduiding op positie 1-2 en de 2 cijfers op positie 13-14 wijzigen.
Dat wil zeggen dat op de posities 3-12 cijfers, hoofdletters of de tekens ‘+’ en ‘*’ kunnen komen.

So I'd say: `(?:NL)?[0-9A-Z+*]{10}[0-9]{2}`